From 362251a0eb540cf2537d78d508b2bcd87fe66321 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:53:54 -0400
Subject: [fs] nfs: allow coalescing of subpage requests

Message-id: <1403798077-28799-36-git-send-email-steved@redhat.com>
Patchwork-id: 84905
O-Subject: [RHEL7.1 PATCH 35/78] nfs: allow coalescing of subpage requests
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Weston Andros Adamson <dros@primarydata.com>

Remove check that the request covers a whole page.

Signed-off-by: Weston Andros Adamson <dros@primarydata.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 6094f83864c1d1296566a282cba05ba613f151ee
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/pagelist.c b/fs/nfs/pagelist.c
index ceb4424..838f7c9 100644
--- a/fs/nfs/pagelist.c
+++ b/fs/nfs/pagelist.c
@@ -921,10 +921,6 @@ static bool nfs_can_coalesce_requests(struct nfs_page *prev,
       !nfs_match_lock_context(req->wb_lock_context,
          prev->wb_lock_context))
    return false;
-  if (req->wb_pgbase != 0)
-   return false;
-  if (prev->wb_pgbase + prev->wb_bytes != PAGE_CACHE_SIZE)
-   return false;
   if (req_offset(req) != req_offset(prev) + prev->wb_bytes)
    return false;
  }
-- 
1.7.1